game-engine/src/main/scala/za/co/entelect/challenge/game/contracts/map/CarGameMap.scala

```diff
@@ -189,4 +191,103 @@ class CarGameMap(players: util.List[Player], mapGenerationSeed: Int, lanes: Int,
   def placeObjectAt(lane: Int, blockNumber: Int, mapObject: Int) = {
     blocks.find(x => x.getPosition().getLane() == lane && x.getPosition().getBlockNumber() == blockNumber).get.mapObject = mapObject;
   }
+
+  def stageFuturePosition(stagedPosition: StagedPosition) ={
+    stagedFuturePositions = stagedFuturePositions.appended(stagedPosition);
+  }
+
+  def resolvePlayerCollisions(): Boolean = {
+    val player1StagedPosition = stagedFuturePositions.find(x => x.getPlayer().getGamePlayerId() == 1).get
+    val player1FuturePosition = player1StagedPosition.getNewPosition()
+
+    val player2StagedPosition = stagedFuturePositions.find(x => x.getPlayer().getGamePlayerId() == 2).get
+    val player2FuturePosition = player2StagedPosition.getNewPosition()
+
+    val playersFuturePositionsAreSame = (player1FuturePosition.getLane() == player2FuturePosition.getLane()) && (player1FuturePosition.getBlockNumber() == player2FuturePosition.getBlockNumber())
+
+    val player1DroveIntoPlayer2 = (player1StagedPosition.getOldPosition().getLane() == player2StagedPosition.getOldPosition().getLane()) && //started off in the same lane
+      (player1StagedPosition.getOldPosition().getBlockNumber() < player2StagedPosition.getOldPosition().getBlockNumber()) && //player 1 was behind
```

Comment:
You could extract these comparisons out to variables, removing the need for comments and making debugging this possible.

To prevent variable spam and variable duplication in this method, you could also extract the logic into a method that takes a LHS player and RHS player and returns LHSdroveIntoRHSPlayer Boolean. Makes this far more readable IMO, but unneeded in terms of functionality.

Reply:
Will definitely add the variables (agree that if I was debugging this next week it would not be easy). Will try the LHS and RHS thing